### PR TITLE
Initial step

### PR DESCRIPTION
**Pull request by phelian23**
In this PR
- Add Student class
- Add Person class
- Add Teacher class

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age, :parent_permission
+
+  def initialize(age, name = 'unknown', parent_permission: true)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
```
Comment (TedLivist) [inline person.rb]: - [ ] Your **`initialize`** method is missing the instance variable for **`@id`**. This **id** can be a randomly generated number between 1 and 1000. Kindly add it.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  def of_age?
+    @age >= 18
+  end
```
Comment (TedLivist) [inline person.rb]: - [ ] Your `of_age?` method should be a private method. Kindly make it so. You can move this method to the bottom of the class and under a `private` keyword

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @phelian23,

Good job so far!
✔️ Good documentation
✔️ Classes are defined in different files
✔️ Student and Teach inherit from Person
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
#### _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-labana — APPROVED** [review]
Hi @phelian23, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**